OpenRA.Mods.Common/WorldExtensions.cs

```diff
@@ -32,8 +32,11 @@ public static IEnumerable<Actor> FindActorsOnLine(this World world, WPos lineSta
 			// Then we iterate over this list, and find all actors for which their health radius is at least within lineWidth of the line.
 			// For actors without a health radius, we simply check their center point.
 			// The square in which we select all actors must be large enough to encompass the entire line's width.
-			var xDir = Math.Sign(lineEnd.X - lineStart.X);
-			var yDir = Math.Sign(lineEnd.Y - lineStart.Y);
+			// xDir and yDir must never be 0, otherwise the overscan will be 0 in the respective direction.
+			var xDiff = lineEnd.X - lineStart.X;
+			var yDiff = lineEnd.Y - lineStart.Y;
+			var xDir = Math.Sign(xDiff != 0 ? xDiff : 1);
```

Comment:
`Math.Sign` is now redundant: this can simplify down to `xDir = xDiff < 0 ? -1 : 1`